qemu/tests/numa_cpu.py

```diff
@@ -122,18 +155,56 @@ def get_hotpluggable_cpus():
         thread_weight = 1
 
     numa_cpu_cli = numa_cpu_cli()
-    specified_cpus = get_hotpluggable_cpus()
-    numa_cpu_monitor = [item[1] for item in vm.monitor.info_numa()]
+    specified_cpus, unspecified_cpus = get_hotpluggable_cpus()
```

Comment:
In this caes, 
specified_cpus = [{0}, {2}, {4}, {6}]
unspecified_cpus = [{'socket_id': 1, 'die_id': 1, 'core_id': 1, 'thread_id': 0}, {'socket_id': 1, 'die_id': 0, 'core_id': 1, 'thread_id': 0}, {'socket_id': 0, 'die_id': 1, 'core_id': 1, 'thread_id': 0}, {'socket_id': 0, 'die_id': 0, 'core_id': 1, 'thread_id': 0}].
Can we use same format for these two variants? 
And for specified_cpus, why is every item a dict? I guess it's just a cpu id?

Reply:
> In this caes,
> specified_cpus = [{0}, {2}, {4}, {6}]
> unspecified_cpus = [{'socket_id': 1, 'die_id': 1, 'core_id': 1, 'thread_id': 0}, {'socket_id': 1, 'die_id': 0, 'core_id': 1, 'thread_id': 0}, {'socket_id': 0, 'die_id': 1, 'core_id': 1, 'thread_id': 0}, {'socket_id': 0, 'die_id': 0, 'core_id': 1, 'thread_id': 0}].
> Can we use same format for these two variants?

Yes, it is international as the doc explaination, they have different uses for the following steps, `specified_cpus` is the cpu ids, and `unspecified_cpus` is the full topology be used by set_numa_node.
To keep same format and avoid repeat conversion, one method is use topology and covert to ids for `specified_cpus` out of this function.   
The difference is return a final or intermediate result for `specified_cpus`.
How about keep it ?

> And for specified_cpus, why is every item a dict? I guess it's just a cpu id?

It is not a dict, it is a tuple, to keep consistent with the format of 'info numa' and guest os output.